core/trino-main/src/main/java/io/trino/operator/join/unspilled/PartitionedLookupSourceFactory.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.join.unspilled;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.SettableFuture;
+import io.trino.operator.join.JoinBridge;
+import io.trino.operator.join.LookupSource;
+import io.trino.operator.join.OuterPositionIterator;
+import io.trino.operator.join.TrackingLookupSourceSupplier;
+import io.trino.spi.type.Type;
+import io.trino.type.BlockTypeOperators;
+
+import javax.annotation.concurrent.GuardedBy;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Supplier;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.util.concurrent.Futures.immediateFuture;
+import static com.google.common.util.concurrent.Futures.immediateVoidFuture;
+import static com.google.common.util.concurrent.Futures.nonCancellationPropagating;
+import static com.google.common.util.concurrent.Futures.transform;
+import static com.google.common.util.concurrent.MoreExecutors.directExecutor;
+import static io.trino.operator.join.OuterLookupSource.createOuterLookupSourceSupplier;
+import static io.trino.operator.join.unspilled.PartitionedLookupSource.createPartitionedLookupSourceSupplier;
+import static java.util.Objects.requireNonNull;
+
+public final class PartitionedLookupSourceFactory
+        implements JoinBridge
```

Comment:
This pretends to be `LookupSourceFactory`, but doesn't implement interface. Name is wrong and confusing